### PR TITLE
Patch moonshine

### PR DESCRIPTION
**Pull request by eustlb**
# What does this PR do?

This PR:
- fixes typos in documentation
- updates expected logits to the ones expected for T4 runners
- fixes badly handled inputs (when no `attention_mask` is provided) in generate's wrap by simply removing the generate wrap 

Justification concerning the last point: it's been discussed that handling per-batch-idx max_lengths is something that is going to be added directly to `generate` (see https://github.com/huggingface/transformers/pull/35676). In the future, `Moonshine`'s processor will return `max_length=[val_1, ... ,val_n]` that will directly be intended to be passed as `max_length` generate's parameter. For this reason, `max_length` should be computed outside generate and therefore the current wrap is not necessary! The wrap was intended to reduce hallucinations that are an edge case but impact evaluation on a test set compare to original codebase. Model cards and recommanded usage will be updated with the correct usage:

```python
from transformers import AutoProcessor, MoonshineForConditionalGeneration
from datasets import load_dataset

processor = AutoProcessor.from_pretrained("UsefulSensors/moonshine-tiny")
model = MoonshineForConditionalGeneration.from_pretrained("UsefulSensors/moonshine-tiny")

ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
inputs = processor([ds[0]["audio"]["array"], ds[0]["audio"]["array"]], return_tensors="pt")
input_values = inputs.input_values

token_limit_factor = 6.5 / processor.feature_extractor.sampling_rate  # Maximum of 6.5 tokens per second
seq_lens = inputs.attention_mask.sum(dim=-1)
max_length = int((seq_lens * token_limit_factor).max().item())

generated_ids = model.generate(input_values, max_length=max_length)
transcription = processor.batch_decode(generated_ids, skip_special_tokens=False)
print(transcription)
```

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
Thanks! As discussed, let's mention why this is needed in the patch!
